### PR TITLE
[FIX] Add ASan linker flags also to libraries.

### PR DESCRIPTION
**Pull request by jpfeuffer**

### CODE REVIEW

**Comment by jpfeuffer** [comment]
Fixes #2684  I think. Wait with merge for my go please. Travis will not catch it.

**Comment by jpfeuffer** [comment]
I spent the whole day looking for generic ways to find the library until I saw the flags are only passed to the executables.. At least I have a backup plan if this Easy solution does not work.

File: cmake/modules/CodeCoverage.cmake [hunk 1]
```diff
@@ -120,53 +122,59 @@ ENDIF() # NOT CMAKE_BUILD_TYPE STREQUAL "Debug"
...
+
+    SET(coverage_info "${CMAKE_BINARY_DIR}/${_outputname}.info")
+    SET(coverage_cleaned "${coverage_info}.cleaned")
+
+    SET(ignorelibpaths \"tests/*\" \"/usr/*\" \"/Applications/*\")
+    foreach(libpath ${_addignorelibpaths})
+        list(APPEND ignorelibpaths \"${libpath}/*\")
+    endforeach()
+
+    ## Workaroud that CMake does not complain during configure, that the log is missing.
```
Comment (hroest) [inline cmake/modules/CodeCoverage.cmake]: typo

Comment (jpfeuffer) [inline cmake/modules/CodeCoverage.cmake]: ah thx

**Comment by jpfeuffer** [comment]
Ok it built through but took more than a day with both enabled at the same time. Coverage was using up all 8GB of RAM. AddressSanitizer is relatively fast.
I googled a bit but nobody else reports such a high RAM usage for Coverage builds with those flags.

**Comment by jpfeuffer** [comment]
We could probably use the flags on less targets and check that no dependencies are tracked but if there are no other ideas, I would leave it that way for now. I will check once more with ONLY coverage.

**Comment by jpfeuffer** [comment]
Ok. Everything works. You have to separate ASan and Coverage, then the runtimes are ok.

**Pull request merged**